Libraries/LibWeb/HTML/Window.cpp

```diff
@@ -1465,6 +1465,10 @@ void Window::scroll(ScrollToOptions const& options)
     x = HTML::normalize_non_finite_values(x);
     y = HTML::normalize_non_finite_values(y);
 
+    // OPTIMIZATION: If we're asked to scroll to (0, 0) and we're already there, do nothing.
+    if (x == 0 && y == 0 && navigable->viewport_scroll_offset().is_zero())
```

Comment:
About both optimizations: why is (0, 0) a special case? Couldn't we compare `scroll_offset()` and `viewport_scroll_offset()` to the requested (x, y) coordinates and no-op them if they're equal?

Reply:
As I understand it, (0, 0) is the only scroll offset that is always guaranteed to be valid. Other values would require performing layout and measuring overflow before we know if we can scroll to that position :)

Reply:
I guess we should clip/limit the scroll region if layout changes, fair. I assumed we'd update our scroll offset variables whenever that happened, but we can't know for certain if that's already the case when this code is invoked.